MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/PolygonSelectToggleActivity.kt

```diff
@@ -0,0 +1,290 @@
+package com.mapbox.mapboxandroiddemo.examples.dds
+
+
+import android.content.Context
+import android.graphics.Color
+import android.graphics.PointF
+import android.os.AsyncTask
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.widget.Toast
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.FeatureCollection
+import com.mapbox.mapboxandroiddemo.R
+import com.mapbox.mapboxsdk.Mapbox
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.style.expressions.Expression.*
+import com.mapbox.mapboxsdk.style.layers.FillLayer
+import com.mapbox.mapboxsdk.style.layers.LineLayer
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory.*
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource
+import kotlinx.android.synthetic.main.activity_dds_polygon_select_toggle.*
+import java.lang.ref.WeakReference
+import java.util.*
+
+/**
+ * Use data-driven styling to style and toggle the colors of various polygons based
+ * on user interaction.
+ */
+class PolygonSelectToggleActivity : AppCompatActivity(), MapboxMap.OnMapClickListener {
+
+    private val BASE_NEIGHBORHOOD_FILL_LAYER_ID = "BASE_NEIGHBORHOOD_FILL_LAYER_ID"
+    private val COLORED_FILL_LAYER_ID = "COLORED_FILL_LAYER_ID"
+    private val LINE_LAYER_ID = "LINE_LAYER_ID"
+    private val NEIGHBORHOOD_POLYGON_SOURCE_ID = "NEIGHBORHOOD_POLYGON_SOURCE_ID"
+    private val PROPERTY_SELECTED = "selected"
+    private val PROPERTY_FILL_COLOR = "fill_color"
+    private val NEIGHBORHOOD_NAME_PROPERTY = "neighborhood_name"
+    private var mapboxMap: MapboxMap? = null
```

Comment:
If you are sure code execution will only occur after mapboxMap is initialized, you can make this lateInit.

Reply:
fixed